xwiki-batchimport-api/src/main/java/org/xwiki/batchimport/internal/DefaultBatchImport.java

```diff
@@ -1369,44 +1375,55 @@ public Document marshalDataToDocumentObjects(DocumentReference pageName, Map<Str
                             } else {
                                 newDocObj.set(key, 0);
                             }
-                            if (addtotags) {
+                            // Whatever the 'honor empty values' parameter is, it never makes sense to add an empty tag
+                            if (addtotags && !StringUtils.isEmpty(stringValue)) {
                                 tagList.add(stringValue);
                             }
                         } else {
                             newDocObj.set(key, value);
 
-                            if (addtotags) {
+                            if (addtotags && !StringUtils.isEmpty(stringValue)) {
                                 tagList.add(stringValue);
                             }
                         }
                     }
                 } else if (key.equals("doc.file")) {
                     // ignore, will be handled by the file function
                 } else if (key.equals("doc.title")) {
-                    newDoc.setTitle((String) value);
+                    // Override the title only if the provided value is not empty or if the import should honor empty
+                    // cells
+                    if (!StringUtils.isEmpty(stringValue) || config.getHonorEmptyValues()) {
+                        newDoc.setTitle((String) value);
+                    }
                 } else if (key.equals("doc.parent")) {
-                    newDoc.setParent((String) value);
+                    if (!StringUtils.isEmpty(stringValue) || config.getHonorEmptyValues()) {
+                        newDoc.setParent((String) value);
+                    }
                 } else if (key.equals("doc.content")) {
-                    newDoc.setContent((String) value);
+                    if (!StringUtils.isEmpty(stringValue) || config.getHonorEmptyValues()) {
+                        newDoc.setContent((String) value);
+                    }
                 }
             }
 
             // set tags, only if needed.
-            // TODO: fix this test here, it should depend on an "empty overwrites" parameter, which should say whether
-            // an empty value is considered significant or not, which should also apply to properties: for now we
-            // overwrite it even with an empty tagList, if the fields for tags is set to something and that something is
-            // void, maybe we shouldn't
-            if (fieldsfortags != null && fieldsfortags.size() > 0) {
+            // NB: we consider that the tags have to be replaced, not added to possibly existing ones, just like
+            // what happens with the other multi-valued properties.
+            if (tagList.size() > 0) {
```

Comment:
I'm not sure I am comfortable with the idea of changing this test here. You can add an extra condition to the original condition to enter this code, but I would like to avoid that all of a sudden (for any reason) tagList may end-up non empty and tag objects start to be added to the documents...
If we want to change the condition and the way the tags are handled, I'd say we'd do that in a different commit for a different issue, this issue is about deciding if empty values from the source file are ignored or not...

Reply:
I removed this code as well.